### ISSUE TITLE
[Good First Issue]: Include HW plugin properties in supported_properties in BATCH Plugin

### ISSUE THREAD

**Issue by zhaixuejun1993** [openvinotoolkit/openvino#23074]
### Context

BATCH Plugin should include HW Plugin compiled_model properties in supported_properties, and get property value from HW plugin compiled_model if user request

benchmark_app -d GPU  -hint throughput -m  <batchable model, mobilenet-v2 for example>
{noformat}
[ INFO ] Model:
[ INFO ]   OPTIMAL_NUMBER_OF_INFER_REQUESTS: 64
[ INFO ]   SUPPORTED_METRICS: OPTIMAL_NUMBER_OF_INFER_REQUESTS SUPPORTED_METRICS NETWORK_NAME SUPPORTED_CONFIG_KEYS EXECUTION_DEVICES
[ INFO ]   NETWORK_NAME: torch-jit-export
[ INFO ]   SUPPORTED_CONFIG_KEYS: AUTO_BATCH_TIMEOUT
[ INFO ]   EXECUTION_DEVICES: OCL_GPU.0
[ INFO ]   AUTO_BATCH_TIMEOUT: 1000{noformat}
The expectation is that it includes GPU properties in SUPPORTED_METRICS.

### What needs to be done?

- [ ] Get properties from HW plugin compiled_model & add to BATCH plugin supported_properties.
u can get HW compiled model properties with m_compiled_model_without_batch & add to the return list. https://github.com/openvinotoolkit/openvino/blob/f9605cd8d4c488ad405397bc4294e35a403ae803/src/plugins/auto_batch/src/compiled_model.cpp#L205
- [ ] Add test.

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/7pVRxUwdWG) - engage in discussions, ask questions and talk to OpenVINO developers

### Contact points

@zhaixuejun1993 @riverlijunjie @ilya-lavrenov @vurusovs @peterchen-intel 

### Ticket

CVS-133381

**Comment by kumar-sanjeeev** [comment]
.take

**Comment by p-wysocki** [comment]
Hello @kumar-sanjeeev, are you still working on this? Is there anything we could help you with?

**Comment by anzr299** [comment]
.take

**Comment by anzr299** [comment]
I am facing some problems running benchmark_app. I followed the tutorials here to set it up: https://docs.openvino.ai/2024/learn-openvino/openvino-samples/benchmark-tool.html.
I cannot find the executable or script in the generated folder as mentioned in the tutorial.

**Comment by anzr299** [comment]
I am having some trouble running benchmark_app. I have followed the tutorial here https://docs.openvino.ai/2024/learn-openvino/openvino-samples/benchmark-tool.html to set it up. After following all the steps, setting up the environment variables etc. I still cant seem to run benchmark_app on cmd

**Comment by anzr299** [comment]
and what is meant by "BATCH plugin supported_properties"?

**Comment by anzr299** [comment]
@p-wysocki

**Comment by anzr299** [comment]
I managed to recreate the results thus far.
I ran benchmark_app with and without the GPU and the results are as follows(Only the part mentioned in this issue):

With GPU:
[Step 8/11] Querying optimal runtime parameters
[ INFO ] Model:
[ INFO ]   OPTIMAL_NUMBER_OF_INFER_REQUESTS: 4
[ INFO ]   NETWORK_NAME: TensorFlow_Frontend_IR
[ INFO ]   EXECUTION_DEVICES: GPU.0
[ INFO ]   AUTO_BATCH_TIMEOUT: 1000


Without GPU:
[ INFO ] Model:
[ INFO ]   NETWORK_NAME: TensorFlow_Frontend_IR
[ INFO ]   OPTIMAL_NUMBER_OF_INFER_REQUESTS: 4
[ INFO ]   NUM_STREAMS: 4
[ INFO ]   AFFINITY: NONE
[ INFO ]   INFERENCE_NUM_THREADS: 16
[ INFO ]   PERF_COUNT: NO
[ INFO ]   INFERENCE_PRECISION_HINT: f32
[ INFO ]   PERFORMANCE_HINT: THROUGHPUT
[ INFO ]   EXECUTION_MODE_HINT: PERFORMANCE
[ INFO ]   PERFORMANCE_HINT_NUM_REQUESTS: 0
[ INFO ]   ENABLE_CPU_PINNING: YES
[ INFO ]   SCHEDULING_CORE_TYPE: ANY_CORE
[ INFO ]   ENABLE_HYPER_THREADING: YES
[ INFO ]   EXECUTION_DEVICES: CPU
[ INFO ]   CPU_DENORMALS_OPTIMIZATION: NO
[ INFO ]   LOG_LEVEL: LOG_NONE
[ INFO ]   CPU_SPARSE_WEIGHTS_DECOMPRESSION_RATE: 1
[ INFO ]   DYNAMIC_QUANTIZATION_GROUP_SIZE: 0
[ INFO ]   KV_CACHE_PRECISION: f16


I am assuming this is the level of detail expected to be printed for the GPU as well?

**Comment by anzr299** [comment]
The changes I am making in compiled_model.cpp of auto_batch plugin are not being reflected when running the benchmark_app command. Does build_samples_msvc not compile the code again?

**Comment by peterchen-intel** [comment]
> The changes I am making in compiled_model.cpp of auto_batch plugin are not being reflected when running the benchmark_app command. Does build_samples_msvc not compile the code again?

The script builds APP only, doesn't build the libraries.

**Comment by anzr299** [comment]
Alright, I have some question regarding what exactly is required in this issue. Is it required to add the compiled model properties from m_compiled_model_without_batch to the supported properties?

**Comment by zhaixuejun1993** [comment]
> Alright, @peterchen-intel, I have a question regarding what exactly is required in this issue. Is it required to add the compiled model properties from m_compiled_model_without_batch to the supported properties?

@anzr299  I think you are right, just add  the compiled model properties from m_compiled_model_without_batch to the supported properties

**Comment by anzr299** [comment]
> > Alright, @peterchen-intel, I have a question regarding what exactly is required in this issue. Is it required to add the compiled model properties from m_compiled_model_without_batch to the supported properties?
> 
> @anzr299 I think you are right, just add the compiled model properties from m_compiled_model_without_batch to the supported properties

Alright, and in regards to testing it, I can build the library and then generate the benchmark app?

**Comment by p-wysocki** [comment]
Hello @anzr299, are you still working on that issue? Do you need any help?

**Comment by anzr299** [comment]
I'm unassigning myself since I am focusing on a smaller subset of problems.

**Comment by p-wysocki** [comment]
No worries @anzr299, come back anytime, we have a lot of issues open. :)

**Comment by awayzjj** [comment]
.take

**Comment by adisan-dev** [comment]
.take

**Comment by p-wysocki** [comment]
Hello @adisan-dev, is there anything we can do to help you with the task? Are you still working on it?

**Comment by adisan-dev** [comment]
Hi @p-wysocki , I was facing some build issues and had also inquired in the Intel Devhub discord. I was able to resolve those and will raise the fix for this issue soon. 
Will ask here if any doubt pops up.

**Comment by Danchi-1** [comment]
.take

**Comment by Aagam-Bothara** [comment]
Hi @Danchi-1  are you still working on this?if not can I take it over?

**Comment by Aagam-Bothara** [comment]
Hi @p-wysocki  is it ok if I take this?

**Comment by Danchi-1** [comment]
Yeah it is. I've been having complications with it and I've not really had
time to go all in

**Comment by Aagam-Bothara** [comment]
Thanks! .take

**Comment by Aagam-Bothara** [comment]
https://github.com/openvinotoolkit/openvino/pull/34441
